compiler/ast.mli

```diff
@@ -15,18 +15,25 @@ type op =    (* Operators *)
   | Div      (* / *)
   | Mod      (* % *)
   | Pow      (* ** *)
+  | Not      (* ! *)
+  | Equal    (* == *)
+  | Neq      (* != *)
+  | Less     (* < *)
+  | Leq      (* <= *)
+  | Greater  (* > *)
+  | Geq      (* >= *)
 
 type expr =                      (* Expressions *)
   | Int_lit of int               (* 42 *)
   | Float_lit of float           (* 42.0 *)
   | String_lit of string         (* "Hello, world" *)
+  | Bool_lit of bool             (* true *)
   | Id of string                 (* x *)
-  | Binop of expr * op * expr    (* a + b *)
-  | Unop of op * expr            (* -5 *)
   | Call of string * expr list   (* add(1, 2) *)
+  | Unop of op * expr            (* -5 *)
+  | Binop of expr * op * expr    (* a + b *)
 
```

Comment:
Why did you change the order here and in generator.ml? Is it a precedence thing?